### PR TITLE
Space Traveler Hub - Completed

### PR DESCRIPTION
**Pull request by davidolanrewaju**
## Space Traveler Hub - Completed

> This PR is a collation of every feature implemented in the complete project

### Implemented Features ♻
- [ ] Installation of project dependencies:
    - React
    - Eslint
    - Stylelint
    - Redux Toolkit
    - react-router-dom 
- [x] Created components for Navigation, Rockets page, Mission page and My Profile page.
- [x] Created redux directory for redux store and reducers. 
- [x] Added reducer to the redux store
- [x] Fetch a list of missions from the [API](https://api.spacexdata.com/v3/missions)
- [x] Display fetched list of missions
- [x] Implemented "Joining Mission" action and "Leaving Mission" action
- [x] Implemented badge switching when the join mission or leave mission button is clicked
- [x] Fetched rocket data from the [API](https://api.spacexdata.com/v4/rockets).
- [x] Display rockets list.
- [x] Implemented rocket booking action and rocket booking cancellation.
- [x] Displayed reserved profile and joined missions on the "My Profile" page

### CODE REVIEW

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

You have do incredible job team, Just a minor  updates to be added. 🙌
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Good job overall following the template design ✅
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
+  "name": "space-traveler-hub",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@reduxjs/toolkit": "^1.9.5",
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
```
[the remark this answers is not in the document]
Comment (davidolanrewaju) [reply package.json:9]: @TadesseDev this wasn't part of the project's requirement

File: src/components/Profile.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const reservedMission = missions.filter((mission) => mission.reserved === true);
+
+  return (
+    <div className="reserved-rockets-con">
+      <div className="reserved-table">
+        <h3>My Rockets</h3>
+        <div className="list-con">
+          {reservedRockets.length === 0 ? (
+            <span>Not reserved rockets!</span>
+          ) : (
```
[the remark this answers is not in the document]
Comment (davidolanrewaju) [reply src/components/Profile.js:18]: @TadesseDev this wasn't part of the design template given for the project. All we did was follow the design template

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (TadesseDev) [reply package.json:9]: Hi @davidolanrewaju, It is actually.  
image

[the remark this answers is not in the document]
Comment (davidolanrewaju) [reply package.json:9]: @TadesseDev Sorry about that... I think I didn't look properly 🙏. Will work on it accordingly

File: src/components/Profile.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidolanrewaju) [reply src/components/Profile.js:18]: @TadesseDev Alright, thanks for clearing that up. We would do our best to implement it.

File: src/tests/Mission.test.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    store.dispatch = jest.fn();
+  });
+
+  test('renders mission data', () => {
+    render(
+      <Provider store={store}>
+        <Mission />
+      </Provider>,
+    );
```
Comment (ishimwezachee) [inline src/tests/Mission.test.js:52]: - [ ]  You have done great job to impliment the tests as suggested by provided by previous reviewers, but there some issue I am facing while I am running them.  check the image to look the issue and fix it . 
image

File: src/tests/Rockets.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+const mockStore = configureStore([]);
+
+describe('Rockets Component', () => {
+  let store;
+
+  beforeEach(() => {
+    const initialState = {
+      rockets: {
+        rockets: [
+          {
```
Comment (ishimwezachee) [inline src/tests/Rockets.test.js:17]: - [ ]  You did a great job, but consider fixing this issue I am facing you tests to proceed  and implement the previous reviewer request properly. 
image

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

image
 ## Highlights
 Good job overall following the template design ✅
 Correct GitFlow ✅
 Good commit history ✅
 Good PR title/description ✅
 All linters passed ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

image
 ## Highlights
 Good job overall following the template design ✅
 Correct GitFlow ✅
 Good commit history ✅
 Good PR title/description ✅
 All linters passed ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Mission.test.js (hunk 3 above)
Comment (davidolanrewaju) [reply src/tests/Mission.test.js:52]: @ishimwezachee please can you explain better what the issue is and an idea or a guide on how to resolve it

**Review by Reem-lab — APPROVED** [review]
# Hi Team,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐⭐

- [ ]  Great job on the funtionality for the setup 💯
- [ ] Great job working on the functionality 🔥
- [ ] Followed Gitflow 🔥
- [ ] Outstanding design ✨✨💯💯
- [ ] No Linters error ✅
- [ ] Kudos on the huge efforts for the tests, All passed ✅✅
- [ ] Professional README file 🔥🔥
- [ ] Kudos for giving the review to each other 🚀🚀

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**